### PR TITLE
Align collector-contrib to use new component layout and fxutil helpers

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

Completes a TODO about using `fxutil.ProvideComponentConstructor` in collector-contrib OTel module. Update the linter by removing more exceptions.

### Motivation

Aligning component definitions now that linters are in place.

### Describe how to test/QA your changes

N/A

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

**Review by jeremy-hanna — APPROVED** [review]
👍

**Review by chouetz — APPROVED** [review]

File: tasks/components.py [hunk 1]
```diff
@@ -169,19 +169,6 @@ def has_type_component(content) -> bool:
...
-]
-
-ignore_package_name = [
-    "comp/trace/agent",
-    "comp/trace/compression",
-    "comp/core/agenttelemetry",
-    "comp/core/log",
-    "comp/core/healthprobe",
-    "comp/core/workloadmeta",
-    "comp/logs/integrations",
```
Comment (GustavoCaso) [inline tasks/components.py:182]: Are all these exceptions fixed? 🎉

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: tasks/components.py (hunk 1 above)
Comment (dustmop) [reply tasks/components.py:182]: Yes, fixed by https://github.com/DataDog/datadog-agent/pull/28039, but I didn't notice in that PR.

**Review by liustanley — APPROVED** [review]

**Pull request merged**